File: NetDeck/CardThumbView.m
//
//  CardThumbView.m
//  Net Deck
//
//  Created by Gereon Steffens on 24.05.14.
//  Copyright © 2016 Gereon Steffens. All rights reserved.
//
//  used in draw simulator & identity picker
//

#import "CardThumbView.h"

@implementation CardThumbView

-(void) awakeFromNib
{
    self.imageView.layer.cornerRadius = 8;
    self.imageView.layer.masksToBounds = YES;
    self.nameLabel.text = nil;
}

-(void) setCard:(Card *)card
{
    self.imageView.image = nil;
    self->_card = card;
    
    [self.activityIndicator startAnimating];
    [self loadImageFor:card];
}

-(void) loadImageFor:(Card*)card
{
    [[ImageCache sharedInstance] getImageFor:card completion:^(Card* card, UIImage* img, BOOL placeholder)
     {
         if ([self.card.code isEqual:card.code])
         {
             [self.activityIndicator stopAnimating];
             
             self.imageView.image = [[ImageCache sharedInstance] croppedImage:img forCard:card];
             self.nameLabel.text = placeholder ? card.name : nil;
         }
         else
         {
             [self loadImageFor:self.card];
         }
     }];
}
@end

File: NetDeck/CardThumbView.swift
//
//  CardThumbView.swift
//  NetDeck
//
//  Created by Gereon Steffens on 30.10.16.
//  Copyright © 2016 Gereon Steffens. All rights reserved.
//

import UIKit

class CardThumbView: UICollectionViewCell {

    @IBOutlet weak var imageView: UIImageView!
    @IBOutlet weak var nameLabel: UILabel!
    @IBOutlet weak var activityIndicator: UIActivityIndicatorView!

    var card = Card.null() {
        didSet {
            self.imageView.image = nil
            self.activityIndicator.startAnimating()
            self.loadImageFor(self.card)
        }
    }
    
    override func awakeFromNib() {
        super.awakeFromNib()
    
        self.imageView.layer.cornerRadius = 8
        self.imageView.layer.masksToBounds = true
        self.nameLabel.text = nil
    }
    
    func loadImageFor(_ card: Card) {
        ImageCache.sharedInstance.getImage(for: card) { (card, img, placerholder) in
            if self.card.code == card.code {
                self.activityIndicator.stopAnimating()
                self.imageView.image = ImageCache.sharedInstance.croppedImage(img, forCard: card)
                self.nameLabel.text = placerholder ? card.name : nil
            } else {
                self.loadImageFor(self.card)
            }
        }
        
    }

}
